tests/Doctrine/Tests/DBAL/Functional/WriteTest.php

```diff
@@ -285,4 +285,44 @@ public function testEmptyIdentityInsert()
 
     }
 
+    /**
+     * @group DBAL-2688
+     */
+    public function testUpdateWhereIsNull()
+    {
+        $this->_conn->insert(
+            'write_table',
+            ['test_int' => '30', 'test_string' => null],
+            ['test_string' => 'string', 'test_int' => 'integer']
+        );
+
+        $data = $this->_conn->fetchColumn('SELECT COUNT(*) FROM write_table WHERE test_int = 30');
+
+        $this->assertEquals(1, $data);
```

Comment:
Right, but the data was an int. Choosing between doing the count in the assert or in the query is CS